cmd/apichecker/apicheck.go

```diff
@@ -0,0 +1,1344 @@
+					resp = append(resp, data[x].Name)
+					data[x].Data.HTMLData.CheckString = checkStr
+				}
+			}
+		}(x)
+	}
+	wg.Wait()
+	file, err := json.MarshalIndent(data, "", " ")
+	if err != nil {
+		return err
+	}
+	for y := range resp {
+		a, err := GetChecklistItems()
+		if err != nil {
+			return err
+		}
+		for z := range a.CheckItems {
+			if strings.Contains(a.CheckItems[z].Name, resp[y]) {
+				err = UpdateCheckItem(a.CheckItems[z].ID, a.CheckItems[z].Name, a.CheckItems[z].State)
+			}
+		}
+	}
+	log.Println(errMap)
+	log.Printf("Following are the exchanges that require updates: %v\n", resp)
+	unsup, err := CheckMissingExchanges(fileName)
+	log.Printf("Following are the exchanges that are supported by GCT but not by apichecker: %v", unsup)
+	return ioutil.WriteFile(fileName, file, 0770)
+}
+
+// CheckChangeLog checks the exchanges which support changelog updates.json
+func CheckChangeLog(htmlData HTMLScrapingData) (string, error) {
+	var dataStrings []string
+	var dataString string
+	var err error
+	switch htmlData.Path {
+	case pathBTSE:
+		dataStrings, err = HTMLScrapeBTSE(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathBitfinex:
+		dataStrings, err = HTMLScrapeBitfinex(htmlData)
+		if err != nil {
+			return "", err
+		}
+	case pathBitmex:
+		dataStrings, err = HTMLScrapeBitmex(htmlData)
+		if err != nil {
+			return "", err
+		}
+	case pathANX:
+		dataStrings, err = HTMLScrapeANX(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathPoloniex:
+		dataStrings, err = HTMLScrapePoloniex(htmlData)
+		if err != nil {
+			return "", err
+		}
+	case pathIbBit:
+		dataStrings, err = HTMLScrapeItBit(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathBTCMarkets:
+		dataStrings, err = HTMLScrapeBTCMarkets(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathEXMO:
+		dataStrings, err = HTMLScrapeExmo(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathBitstamp:
+		dataStrings, err = HTMLScrapeBitstamp(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathHitBTC:
+		dataStrings, err = HTMLScrapeHitBTC(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathBitflyer:
+		dataStrings, err = HTMLScrapeBitflyer(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathLakeBTC:
+		dataStrings, err = HTMLScrapeLakeBTC(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathKraken:
+		dataStrings, err = HTMLScrapeKraken(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathAlphaPoint:
+		dataStrings, err = HTMLScrapeAlphaPoint(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
```

Comment:
Every instance in this switch where you return `dataStrings[0]`, dataStrings only contains one entry. Is this ever different?
Because if its only ever one you can neaten up this switch case
```
switch (blah) {
case pathAlphaPoint:
		dataStrings, err = HTMLScrapeAlphaPoint(htmlData)
case blah:
               dataStrings, err = HTMLScrapeAlphaPoint(htmlData)
}
if err != nil {
    return "", err
}
switch {
	case len(dataStrings) == 1:
....
```

Reply:
some cases only return 1 date where as others may return multiple. so the ones with singular check string dont need any more work to be done on them so can just return right there. sorry if i misinterpreted what u meant